src/hnsw/utils.c

```diff
@@ -33,7 +36,77 @@ void PopulateUsearchOpts(Relation index, usearch_init_options_t *opts)
     opts->connectivity = HnswGetM(index);
     opts->expansion_add = HnswGetEfConstruction(index);
     opts->expansion_search = HnswGetEf(index);
-    opts->metric_kind = usearch_metric_l2sq_k;
+    opts->metric_kind = HnswGetMetricKind(index);
     opts->metric = NULL;
     opts->quantization = usearch_scalar_f32_k;
 }
+
+int IsInsideQuotes(const char *query, int op_start, int op_end)
+{
+    if(op_start == 0) return 0;
+
+    int    single_quote_start = -1;
+    size_t offset = 0;
+
+    while(strcmp(query + offset, "\0") != 0) {
+        char s = *(query + offset);
+        if(s == '\'') {
+            // if current char is single quote
+            if(single_quote_start == -1) {
+                // if we didn't encounter any unclosed single quote before
+                // keep the start offset
+                single_quote_start = offset;
+            } else {
+                // if we already have open single quote
+                // check if our operator is inside quotes
+                // return 1
+                if(single_quote_start < op_start && offset > op_end) {
+                    return 1;
+                }
+                single_quote_start = -1;
+            }
+        }
+        offset += 1;
+    }
+
+    return 0;
+}
+
+int CheckOperatorUsage(const char *query)
+{
+    const char *pattern = "(<->)";
+    const char *orderby_pattern = "order by";
+    regex_t     regex;
+    regex_t     regex2;
+    regmatch_t  matches[ 1 ];
+    int         reti;
+    int         reti2;
+    int         status = 0;
+
+    reti = regcomp(&regex, pattern, REG_EXTENDED | REG_ICASE);
+    reti2 = regcomp(&regex2, orderby_pattern, REG_EXTENDED | REG_ICASE);
+    if(reti || reti2) {
+        elog(ERROR, "Could not compile regex");
+        return status;
+    }
+
+    // Find all occurrences of the <-> operator
+    int offset = 0;
+    while((reti = regexec(&regex, query + offset, 1, matches, 0)) == 0) {
+        long start = offset + matches[ 0 ].rm_so;
+        long end = offset + matches[ 0 ].rm_eo;
+        char substring[ start - offset ];
+        strncpy(substring, query + offset, start - offset);
```

Comment:
Should substring have length start-offset+1 and a null character in the end?

Say offset = 0, start=4
substring is 4 characters long.
strncpy copies start-offset = 4 characters into substring
Substring is pased into regexec shich does not take a length argument so i assume it expects a null terminated string. But what we passed in is not a null terminated string.

Reply:
You're right, it may make overflows, I will ensure that substring is null terminated before passing to regexec